modules/glue-catalog-table/variables.tf

```diff
@@ -34,12 +34,12 @@ variable "parameters" {
 }
 
 variable "partition_index" {
```

Comment:
A way we've handled breaking changes before is to create a new variable `partition_indices` and move the existing one to `variables-deprecated.tf`.  Then, if a value exists for `partition_index`, use it to set a value in the new map variable.

Reply:
makes sense, I will  apply the requested changes

Reply:
I've added a new local var as following

```hcl
  partition_indices = var.partition_index != null ? merge(var.partition_indices, {
    "0" = {
      index_name = var.partition_index.index_name
      keys       = var.partition_index.keys
    }
  }) : var.partition_indices
```

It basically merges `partition_index` if its defined with `partition_indices` otherwise returns `partition_indices` as is and also moved the deprecated var as you suggested to `variables-deprecated.tf`.

I have tested it as follows 

```hcl
module "glue_catalog_table" {
...

  partition_index = {
    index_name = "index_organization_id_event_id"
    keys       = [ "organization_id", "event_id"]

  }
  
  partition_indices = {
    "index_year_month_day" = {
      index_name = "index_year_month_day"
      keys       = ["year", "month", "day"]
    }
  }

  partition_keys = ...

  storage_descriptor = ...
```

and plan generates both indices as follows

```hcl

...

      + partition_index {
          + index_name   = "index_organization_id_event_id"
          + index_status = (known after apply)
          + keys         = [
              + "organization_id",
              + "event_id",
            ]
        }
      + partition_index {
          + index_name   = "index_year_month_day"
          + index_status = (known after apply)
          + keys         = [
              + "year",
              + "month",
              + "day",
            ]
        }

...
```